knowledge-base/editor-add-icons.md

```diff
@@ -0,0 +1,209 @@
+---
+title: Add Font Icons in Editor
+description: How to allow inserting <i> tags in the Editor for Blazor to render custom icons in the Editor content?
+type: how-to
+page_title: Add Icons in Editor
+slug: editor-kb-add-icons
+position: 
+tags: blazor, editor, prosemirror, schema, icon
+ticketid: 
+res_type: kb
+---
+
+## Environment
+<table>
+	<tbody>
+		<tr>
+			<td>Product</td>
+			<td>Editor for Blazor</td>
+		</tr>
+	</tbody>
+</table>
+
+
+## Description
+
+This KB article answers the following questions:
+
+* How to add custom icons in the Editor content? When adding an icon such as `<i class="fa-light fa-envelope">`, the icon's HTML disappears upon saving.
+* How can I add an `<i>` element inside the Editor?
+
+## Solution
+
+The default schema of the Telerik Editor does not include an `<i>` tag, so the Editor strips such elements automatically. To allow adding font icons through `<i>` tags in the Editor content:
+
+1. [Modify the ProseMirror schema]({%slug editor-modify-default-schema%}) to include a node for the `<i>` element.
+2. Ensure the required font icon stylesheets can affect the Editor content. The approach varies depending on the [edit mode of the Editor]({%slug editor-edit-modes-overview%}):
+    * [Iframe Edit Mode](#add-icons-in-iframe-edit-mode)
+    * [Div Edit Mode](#add-icons-in-div-edit-mode)
+
+
+### Add Icons in Iframe Edit Mode
+
+When the [Editor `EditMode` is set to `EditorEditMode.Iframe`]({%slug editor-edit-modes-iframe%}), the content area is inside an `<iframe>` element that does not apply the CSS rules from the current page.
+
+This means that you need to inject the icons stylesheet into the `<iframe>`, so the icons are properly rendered. At the time of writing (UI for Blazor **6.1.0**), [the Editor does not support injecting your CSS files into the Iframe](https://feedback.telerik.com/blazor/1543925-add-the-ability-to-inject-css-files-into-the-iframe) but you can inject them with JSInterop in the `OnAfterRenderAsync`.
+
+>caption Add icons in an Editor with Iframe edit mode
+
+````CSHTML
+@using Telerik.Blazor.Components.Editor
+@inject IJSRuntime js
+
+<TelerikEditor @bind-Value="@EditorValue"
+               Tools="@EditorTools"
+               Schema="schemaProvider"
+               Width="600px"
```

Comment:
The Editor `Width` may be larger than the width of the REPL box, e.g. on phones. We can skip it unless necessary for some reason.